src/outputs/restart.hpp

```diff
@@ -0,0 +1,165 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef OUTPUTS_RESTART_HPP_
+#define OUTPUTS_RESTART_HPP_
+//! \file io_wrapper.hpp
+//  \brief defines a set of small wrapper functions for MPI versus Serial Output.
+
+#ifdef HDF5OUTPUT
+#include <hdf5.h>
+#endif
+
+#include <cinttypes>
+#include <string>
+#include <vector>
+
+#include "outputs/parthenon_hdf5.hpp"
+namespace parthenon {
+class Mesh;
+
+class RestartReader {
+ public:
+  explicit RestartReader(const char *theFile);
+
+  // Gets single block data for variable name and
+  // fills internal data for given pointer
+  // returns 1 on success, -1 on failure
+  //! \fn void RestartReader::ReadBlock(const char *name, const int blockID, T *data)
+  //  \brief Reads data for one block from restart file
+  template <typename T>
+  int ReadBlock(const char *name, const int blockID, T *data, size_t vlen) {
+    IndexRange range{blockID, blockID};
+    return ReadBlocks(name, range, data, vlen);
+  }
+
+  // Gets data for all blocks on current rank.
+  // Assumes blocks are contiguous
+  // fills internal data for given pointer
+  // returns NBlocks on success, -1 on failure
+  template <typename T>
+  int ReadBlocks(const char *name, IndexRange range, T *data, size_t vlen = 1) {
+#ifdef HDF5OUTPUT
+    try {
+      herr_t status;
+
+      // compute block size, probaby could cache this.
+      hid_t theHdfType = getHdfType(data);
+
+      hid_t dataset = H5Dopen2(fh_, name, H5P_DEFAULT);
+      if (dataset < 0) {
+        return -1;
```

Comment:
Again, non-blocking. But I want these to be`enum`s.

Reply:
You're like a bulldog, once you sink your teeth in there, you refuse to let go...

Reply:
**woof** *No magic numbers!* **woof**

Reply:
I'm going to defer this one.  
However I do agree with you.

Reply:
Yeah that's fine.